### PR TITLE
Update Qwen2.5 configs

### PR DESCRIPTION
**Pull request by joecummings**
1. I turned activation checkpointing off for all 0.5B models and for 1.5B LoRA models. No point. 
2. I turned on memory logging

Everything else is cosmetic.

### CODE REVIEW

File: recipes/configs/qwen2_5/0.5B_full.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#
+# To launch on 2 devices, run the following command from root:
+#   tune run --nproc_per_node 2 full_finetune_distributed --config qwen2_5/0.5B_full
+#
+# You can add specific overrides through the command line. For example
+# to override the checkpointer directory while launching training:
+#   tune run --nproc_per_node 2 full_finetune_distributed --config qwen2_5/0.5B_full checkpointer.checkpoint_dir=<YOUR_CHECKPOINT_DIR>
+#
+# This config works best when the model is being fine-tuned on 2+ GPUs.
+# Single device full finetuning requires more memory optimizations. It's
```
Comment (ebsmothers) [inline recipes/configs/qwen2_5/0.5B_full.yaml:15]: nit: not really true in this case

File: recipes/configs/qwen2_5/0.5B_full_single_device.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#
+# You can add specific overrides through the command line. For example
+# to override the checkpointer directory while launching training:
+#   tune run full_finetune_single_device --config qwen2_5/0.5B_full_single_device checkpointer.checkpoint_dir=<YOUR_CHECKPOINT_DIR>
+#
+# This config works only for training on single device.
+
+# Model arguments
+model:
+  _component_: torchtune.models.qwen2_5.qwen2_5_0_5b
```
Comment (ebsmothers) [inline recipes/configs/qwen2_5/0.5B_full_single_device.yaml:18]: I understand that you're making the filename change 0_5B -> 0.5B for consistency with other configs, but honestly would prefer to just move everything to 0_5B so it matches the builders

**Review by calvinpelletier — COMMENTED** [review]
I thought we use underscores instead of periods?: https://github.com/pytorch/torchtune/pull/1863#discussion_r1806701325

File: recipes/configs/qwen2_5/1.5B_full.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#
+# To launch on 2 devices, run the following command from root:
+#   tune run --nproc_per_node 2 full_finetune_distributed --config qwen2_5/1.5B_full
+#
+# You can add specific overrides through the command line. For example
+# to override the checkpointer directory while launching training:
+#   tune run --nproc_per_node 2 full_finetune_distributed --config qwen2_5/1.5B_full checkpointer.checkpoint_dir=<YOUR_CHECKPOINT_DIR>
+#
+# This config works best when the model is being fine-tuned on 2+ GPUs.
+# Single device full finetuning requires more memory optimizations. It's
```
Comment (ebsmothers) [inline recipes/configs/qwen2_5/1.5B_full.yaml:15]: prob same here: 1.5 * 8 = 12, so most users will not need any optimizations to run this

File: recipes/configs/qwen2_5/1.5B_full_single_device.yaml [hunk 4]
```diff
@@ -1,108 +1,75 @@
...
+output_dir: /tmp/Qwen2.5-1.5B-Instruct-finetune
 metric_logger:
   _component_: torchtune.training.metric_logging.DiskLogger
-  log_dir: ${output_dir}
-output_dir: /tmp/Qwen2_5-1_5B-Instruct-finetune
+  log_dir: ${output_dir}/logs
 log_every_n_steps: 1
-log_peak_memory_stats: False
-
-# Profiler (disabled)
```
Comment (ebsmothers) [inline recipes/configs/qwen2_5/1.5B_full_single_device.yaml:85]: controversial take but if we want consistency we should leave these in. idc too much for this PR but I thought that was the whole point of a bunch of @felipemello1's changes. Either way would like to compress this config substantially in a separate PR

**Review by ebsmothers — APPROVED** [review]
I hate decimal points

**Comment by joecummings** [comment]
> I thought we use underscores instead of periods?: [#1863 (comment)](https://github.com/pytorch/torchtune/pull/1863#discussion_r1806701325)

Yeah this is a misleading comment. We do use underscores for model builders, but the model should just get downloading to a directory with the exact same name as the model on the Hub.

File: recipes/configs/qwen2_5/1.5B_full_single_device.yaml (hunk 4 above)
Comment (felipemello1) [reply recipes/configs/qwen2_5/1.5B_full_single_device.yaml:85]: I am ok with making the profiler simpler is a separate PR

File: recipes/configs/qwen2_5/0.5B_full_single_device.yaml (hunk 2 above)
Comment (felipemello1) [reply recipes/configs/qwen2_5/0.5B_full_single_device.yaml:18]: for llama3.2 we added 3.2 to the path like you did

image

Comment (joecummings) [reply recipes/configs/qwen2_5/0.5B_full_single_device.yaml:18]: This is the same thing we have here, no?

**Pull request merged**